plugins/hls-overloaded-record-dot-plugin/src/Ide/Plugin/OverloadedRecordDot.hs

```diff
@@ -105,7 +117,14 @@ instance Hashable CollectRecordSelectors
 instance NFData CollectRecordSelectors
 
 data CollectRecordSelectorsResult = CRSR
-    { recordInfos       :: RangeMap RecordSelectorExpr
+    { -- |We store everything in here that we need to create the unresolved
+      -- codeAction, the range, an uniquely identifiable int, and the expression
+      -- for the selector expression that we use to generate the name
+      records           :: RangeMap (Int, HsExpr (GhcPass 'Renamed))
+      -- |This is for when we need to fully generate a textEdit. It contains the
+      -- whole expression we are interested in indexed to the unique id we got
+      -- from the previous field
+    , recordInfos       :: IntMap.IntMap RecordSelectorExpr
```

Comment:
The logic here is a bit funny to me: for the `HsExpr` we put it in the result of the `RangeMap`, but the `RecordSelectorExpr` is in a separate map! I would expect something like:
```
records :: RangeMap Int
recordInfo :: IntMap (HsExpr, RecordSelectorExpr)
```

Reply:
That is what I originally had, but then I had to query the map, later on, to extract the selector expression because we need that for the name. Now I put the selector expression, which is the part I need for the name, in the range map, and then with one lookup to the range map I can build the first code action, and then I only need to do one lookup to the int map when I want to actually resolve the code action.

Reply:
Okay, maybe just try to explain that a bit more in the comment? Basically the first bit is what we need for the initial code action, and the second bit is what we need for resolve.